api/v1alpha1/onload_types.go

```diff
@@ -4,25 +4,76 @@
 package v1alpha1
 
 import (
+	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+// Currently unimplemented
+type OnloadBuildSpec struct {
+}
+
+// Currently unimplemented
+type DevicePluginBuildSpec struct {
+}
+
 // OnloadSpec defines the desired state of Onload
 type OnloadSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// KernelImage
+	// An image that contains the out-of-tree kernel modules used by onload,
+	// including the sfc driver.
+	KernelImage string `json:"kernelImage"`
```

Comment:
Do we want multiple kernel versions like `regexp` in https://kmm.sigs.k8s.io/documentation/deploy_kmod/ to (1) support the node kernel upgrade more seamlessly and (2) allow the selector to select nodes with different kernel versions?